○×ゲーム設計書とソースコードの修正

KanayamaHidekazu opened the pull request:
@JunjiHanashiro
新人三名で○×ゲームの要件定義書を作成しましたので、レビューをお願いいたします。

KanayamaHidekazu committed:
○×ゲームの要件定義書を作成しました

KanayamaHidekazu committed:
インターフェース定義書とテーブル定義書を追加しました

KanayamaHidekazu commented:
@JunjiHanashiro
インターフェース定義書とテーブル定義書を作成しましたので、レビューお願いします

KanayamaHidekazu committed:
テーブル定義書のテーブル構成を変更いたしました。

KanayamaHidekazu commented:
@JunjiHanashiro
テーブル定義書の構成を変更しましたので、確認をお願いします。

KanayamaHidekazu committed:
要件定義書・テーブル定義書を更新しました。
システム構成図・機能概要書を追加しました。

KanayamaHidekazu commented:
@JunjiHanashiro 
要件定義書・テーブル定義書を更新、システム構成図・機能概要書を追加しましたので、確認をお願いします。

KanayamaHidekazu committed:
機能概要書を更新しました。

KanayamaHidekazu commented:
@JunjiHanashiro 
機能概要書を更新しましたので、確認をお願いします。

KanayamaHidekazu committed:
インターフェース定義書を更新しました

KanayamaHidekazu commented:
@JunjiHanashiro 
インターフェース定義書を更新しましたので、確認お願いします。

KanayamaHidekazu committed:
インターフェース定義書を修正しました。

KanayamaHidekazu commented:
@JunjiHanashiro 
インターフェース定義書を修正しましたので確認お願いします。

KanayamaHidekazu committed:
機能概要書の修正を行いました。

KanayamaHidekazu commented:
@JunjiHanashiro 
機能概要書の修正を行いましたので確認お願いします。

ptechatsugai committed:
要件定義書・テーブル定義書・システム構成図を最新の状態にしました。

ptechatsugai commented:
@JunjiHanashiro 
要件定義書・テーブル定義書・システム構成図を最新のものに更新しました。確認お願いします。

KanayamaHidekazu committed:
サーバー機能のクラス図とシーケンス図を作成しました。

KanayamaHidekazu committed:
Merge branch 'truefalseBranch' of https://github.com/kashima1115/TrueOrFalseGame into truefalseBranch

ptechatsugai committed:
クライアント機能のシーケンス図を追加しました。

ptechatsugai committed:
クライアント機能のソースを追加しました。

tec-arahari committed:
servlet jsp commit

KanayamaHidekazu committed:
サーバー機能のソースとjavadocを作成しました

KanayamaHidekazu commented:
@JunjiHanashiro 
各機能のソースコードとjavadocを追加しましたので、確認をお願いします。

ptechatsugai committed:
config.propertiesにMessageQueueの設定（相手のIPアドレス）を追加しました。
また、新たに作成されたJavaDocのフォルダ内にクライアント機能のJavaDocを追加しました。

tec-arahari commented on 06.ソース/server/BattleIdAdmin.java:
```diff
@@ -0,0 +1,34 @@
+ *
+ */
+public class BattleIdAdmin {
+	private int battleId;
```
@KanayamaHidekazu 
mous

ptechatsugai committed:
Brainの情報を利用できない問題を解決しました。また、BrainBeanをjar内のみにしました。

KanayamaHidekazu committed:
ActiveMQ周りのソースを更新しました

KanayamaHidekazu committed:
Merge branch 'truefalseBranch' of https://github.com/kashima1115/TrueOrFalseGame into truefalseBranch

ptechatsugai committed:
brain.jarの仕様を変更しました。AccessBrainThreadMode（brainのタイムアウトエラーに対応）を追加しました。javadocを更新しました。

tec-arahari committed:
servlet jsp context commit

tec-arahari commented:
@JunjiHanashiro 
先日のレビュー時に指摘された無駄な空白の削除とdata sourceを使用したDB接続の変更を行ったものをコミット致しましたのでレビューをお願い致します。

なお、接続設定をserver.xmlから取り込むのに難航しており、現状META-INF下に配置したcontext.xmlを使用しております。
当ファイルもコミットしているので合わせてご査収ください。

KanayamaHidekazu committed:
サーバー機能のソースを最新のものに更新しました

KanayamaHidekazu commented:
@JunjiHanashiro 
先日のレビューで指摘された、変数の宣言箇所の修正と、DBにアクセスするクラスのコネクションを作成するメソッドと切断するメソッドを作成しました。
また、その他機能の複数のソースを追加しました。

ptechatsugai committed:
ConvertJSONにおいてJSONObjectに入れる際のキー名をインターフェース定義書の仕様と合致するように修正。SequenceControlでIPアドレスの取得方法を変更。

ptechatsugai commented:
@JunjiHanashiro 
金山さんの指摘によりJSONObjectに入れるキー名が定義書と異なっている問題が発覚しConvertJSONを修正しました。
SequenceControl内でIPアドレスを特定する方法を変更しました。

KanayamaHidekazu committed:
ソースに直接、値を記述していた部分を、定数に置き換えました

tec-arahari committed:
add bean PreparedStatement

tec-arahari commented:
@JunjiHanashiro 

サーブレットのBeanを増やし、SQL文をプリペアドステートメントに変更致しました。
レビューをお願い致します。

ptechatsugai committed:
AccessBrainの指摘のあった箇所を修正しました。それに伴う他のクラスの修正も行いました。

ptechatsugai commented:
@JunjiHanashiro 
指摘事項のあったAccessBrainの修正をしました。また、AccessBrainに関連したクラスのコードも修正しました。

KanayamaHidekazu committed:
DBアクセス関連処理の修正とテストコードの追加を行いました

KanayamaHidekazu commented:
@JunjiHanashiro 
8/31のレビューで指摘された、サーバーのDBアクセス関連についての指摘点の対応とテストコードの作成をしました

ptechatsugai committed:
createBrainの実行をSequenceControl上で行うよう修正しました。テストコードを新たに作成しました。

tec-arahari committed:
back up commit

ptechatsugai committed:
テストコードを少し変更しました。また、一部ソースを修正しました。

KanayamaHidekazu committed:
ロジック情報をListで管理していたものを、Mapに変更しました。
また、それに伴ってソースを修正しました

KanayamaHidekazu commented:
@JunjiHanashiro 
サーバー機能に関して、ロジック情報をListで管理していたものをMapに変更しました。
それに伴って、ソースを修正しました。

ptechatsugai committed:
ソースコードを簡潔かつ効率的なものになるよう修正しました。また、brainのソースコードも追加しました。

tec-arahari committed:
add Utill

KanayamaHidekazu committed:
試合管理クラスのソースを複数のメソッドに分割するリファクタを行いました。
それに伴う、クラスの追加、テストコードの修正を行いました

ptechatsugai committed:
SequenceControlのリファクタを行いました。テストコードを「06.ソース」内の「test」に格納しました。Brain.jarの元となったソースコードを「06.ソース」の「brain」に格納しました。
ConvertJSONにて配列の処理に問題があったため修正しました。

KanayamaHidekazu committed:
DBのテーブル定義が変更したことに伴って、サーバーのソースを変更しました

tec-arahari committed:
new jar file

tec-arahari committed:
Merge branch 'truefalseBranch' of https://github.com/kashima1115/TrueOrFalseGame into truefalseBranch

tec-arahari commented:
@JunjiHanashiro 

試合日時をDateTime型に変更しました。

さらに、必要なjarファイルをまとめるフォルダを作成しました。
今回コミットした「joda-time-2.9.9.jar」はWEB-INF直下に配置をお願い致します。

KanayamaHidekazu committed:
プロジェクトに利用しているjarファイルを追加しました

tec-arahari committed:
sample DB table

tec-arahari committed:
Merge branch 'truefalseBranch' of https://github.com/kashima1115/TrueOrFalseGame into truefalseBranch

KanayamaHidekazu committed:
BattleAdmin.javaのソースを更新しました

ptechatsugai committed:
ActiveMQのActiveMQConnectionFactoryの中身を変更しました。

KanayamaHidekazu committed:
ClientAddressBean.javaを削除しました

KanayamaHidekazu committed:
Merge branch 'truefalseBranch' of https://github.com/kashima1115/TrueOrFalseGame into truefalseBranch

KanayamaHidekazu committed:
テストコードを追加しました

tec-arahari committed:
web.xml

tec-arahari committed:
jsp servlet modify

KanayamaHidekazu committed:
試合管理クラスとロジック情報管理クラスのテストコードを追加しました。
また、試合管理クラスの修正を行いました。

ptechatsugai committed:
ActiveMQ使用時、メッセージを読み込めない問題を修正しました。

KanayamaHidekazu committed:
プロジェクトで使用するjarファイルを追加しました

ptechatsugai committed:
client機能のソースコードを最新のものに更新しました。

ptechatsugai committed:
Merge branch 'truefalseBranch' of https://github.com/kashima1115/TrueOrFalseGame into truefalseBranch

KanayamaHidekazu committed:
9/13レビューで指摘されたメソッドの分割のソースを修正中のものをコミットしました。
また、ロジック情報管理クラスのテストコードを追加しました。

KanayamaHidekazu committed:
Merge branch 'truefalseBranch' of https://github.com/kashima1115/TrueOrFalseGame into truefalseBranch

tec-arahari committed:
new SequenceControlTest.java

tec-arahari committed:
送信するメッセージのテストに受信したメッセージのテストを追加

ptechatsugai committed:
EventType・SequenceControlForTest・ActiveMQMessagingForTestを新しく作成しました。
JavaDocを更新しました。
ソース内に不適切な記述があったため修正しました。

KanayamaHidekazu committed:
花城さん指摘点の修正と、テストコードを追加しました。
その他、ソースの微修正しました。

tec-arahari committed:
Brainのテストコードを追加しました

KanayamaHidekazu committed:
BattleAdmin.javaのリファクタリングとそれに伴う微修正を行いました。
また、LocationAdminのテストクラスを作成しました。

ptechatsugai committed:
ServerActiveMQMessagingクラスの設定等をActiveMQMessagingクラスと共通化しました。

ptechatsugai committed:
ActiveMQの送受信について一部コードを分かりやすくなるように変更しました。

ptechatsugai committed:
テストを行いやすくなるように修正しました。

ptechatsugai committed:
先のコミットの追加です。

ptechatsugai committed:
EventTypeを修正しました。

tec-arahari committed:
ダミーを使用したclientJunitテスト追加

ptechatsugai committed:
例外処理の変更やメンバ変数のカプセル化・ログの記述変更などを行いました。

KanayamaHidekazu committed:
BattleAdmin.javaのリファクタ及び、それに伴うテストクラスの修正を行いました。
また、その他のクラスの微修正を行いました。

tec-arahari committed:
盤面が反転していた問題を修正しました

ptechatsugai committed:
テスト元の例外処理変更に伴いテストコードも修正しました。

tec-arahari committed:
clientの変更に伴った修正を行いました。

KanayamaHidekazu committed:
サーバーの一部クラスをstaticメソッドに変更しました。
アクセス修飾詞を見直しました。
DBにアクセスするクラスとエラー通知クラスのテストクラスを追加しました。
ServerTry.javaを削除しました。
その他微修正しました。

ptechatsugai committed:
log4j.xmlを追加しました。ActiveMQでもlog4jが使われているため、プログラム実行時はこのlog4j.xmlが先に読み込まれるように設定をお願いします。

KanayamaHidekazu committed:
BattleAdmin.javaの微修正を行いました。

KanayamaHidekazu committed:
BattleAdmin.javaの試合終了判定の条件分岐について、処理が重複していた箇所が存在していたので、これを1つに統合しました。
一部メソッドをstaticメソッドに変更しました。
その他、微修正を行いました。

KanayamaHidekazu committed:
一部クラスの名称を変更しました。
一部クラスのメソッド名を変更しました。
BattleFlow.java(BattleAdmin.java)の試合継戦・終了の条件分岐を修正しました。
InformError.javaのメソッドをstaticメソッドにしました。
InformError.javaの通知オブジェクトを格納する配列を削除しました。
その他微修正を行いました。

ptechatsugai committed:
例外処理の仕様変更に対応しました。
すでに使われていないため、ActiveMQMessagingForTestは削除してください。

KanayamaHidekazu committed:
試合の続行・終了判定が働かない不具合を修正しました

ptechatsugai committed:
昨日更新したConvertJSONにて修正漏れがあったため訂正しました。

KanayamaHidekazu committed:
結合テストを行い、発生した不具合の修正を行いました。

KanayamaHidekazu committed:
MessageQueueControllerインターフェース、ServerActiveMQMessaging.javaの例外処理を修正しました。
CheckJsonKey.javaのテストクラスを追加しました。
一部テストクラスの名称を変更しました。
その他、微修正を行いました。

ptechatsugai committed:
サーバー側のキュー名変更を反映しました。

ptechatsugai committed:
manualのbrainを自分の駒が○か×か分かるようにしました。

KanayamaHidekazu committed:
○×ゲームプログラムで使用するDB内のデータを追加しました。

tec-arahari committed:
今週の成果物です

KanayamaHidekazu committed:
DBUnitに関するjarを追加しました
DbOperationTestクラスをDBUnitを使用したソースに変更しました（作成中）
DBに挿入するテストデータを記述したDbSet.xmlを追加しました。

KanayamaHidekazu committed:
Merge branch 'truefalseBranch' of https://github.com/kashima1115/TrueOrFalseGame into truefalseBranch

ptechatsugai committed:
試合結果一覧で勝敗が正しく表示されない問題が修正されました。

KanayamaHidekazu committed:
サーバーのテストクラスを更新しました

KanayamaHidekazu committed:
Merge branch 'truefalseBranch' of https://github.com/kashima1115/TrueOrFalseGame into truefalseBranch

ptechatsugai committed:
同じBrain同士で戦った場合の動きを再現するテストを追加しました。
ただし、勝敗のみをチェックするようにしてあるので異常動作については各自で確認お願いします。

ptechatsugai committed:
Merge branch 'truefalseBranch' of https://github.com/kashima1115/TrueOrFalseGame into truefalseBranch

tec-arahari committed:
作成したBrainを手入力のBrainと試合を行えるテストコード
brain2をクラスパスに追加してください。

KanayamaHidekazu committed:
DbOperation.javaにおいて、レコードが無いときにmax関数がnullから0を返すように修正しました。
DbOperationTestにおいて、テストを行ってもDBの各テーブルのauto_incrementの値がテスト前と変わらないようにソースを修正しました。

tec-arahari committed:
制限時間エラー処理を追加いたしました

ptechatsugai committed:
ServerActiveMQMessagingの仕様変更に対応しました。

KanayamaHidekazu committed:
DbOperationTestにおいて、事前準備・事後処理の構成を変更しました。
プログラムに不要なクラスを削除しました。

KanayamaHidekazu committed:
DbOperationTestにおいて、DBの指し手テーブルから無効な試合IDを持つレコードを削除するメソッドのテストコードを追加しました。

ptechatsugai committed:
資料を実装にあわせたものに修正しました。

tec-arahari commented:
@JunjiHanashiro 
 設計書、仕様書のブラッシュアップを行いました。
よろしければレビューをお願い致します。

ptechatsugai committed:
シーケンス図を追加しました。

ptechatsugai commented:
@JunjiHanashiro 
シーケンス図のブラッシュアップを行いましたので、よろしければレビューをお願い致します。

KanayamaHidekazu committed:
○×ゲームプログラム（サーバープログラム）を起動するbatファイルを作成しました

ptechatsugai committed:
クライアントプログラムをバッチで起動できるようになりました。ActiveMQは各自で起動してください。

KanayamaHidekazu committed:
activemq.xmlの設定を変更しました

KanayamaHidekazu committed:
serverのバッチの余計なソースが混じっていたので削除しました
